common/src/main/java/org/apache/cassandra/sidecar/common/utils/TimeUtils.java

```diff
@@ -0,0 +1,36 @@
+package org.apache.cassandra.sidecar.common.utils;
+
+import java.time.Duration;
+import java.util.concurrent.ThreadLocalRandom;
+
+/**
+ * Utility class for manipulating dates, times, and durations
+ */
+public final class TimeUtils {
+    /**
+     * Private constructor that prevents unnecessary instantiation
+     *
+     * @throws IllegalStateException when called
+     */
+    private TimeUtils()
+    {
+        throw new IllegalStateException(getClass() + " is a static utility class and shall not be instantiated");
+    }
+
+    /**
+     * Returns a random duration with millisecond precision that is uniformly distributed between two provided durations
+     *
+     * @param minimum minimum possible duration, inclusive
+     * @param maximum maximum possible duration, inclusive
+     *
+     * @return random duration uniformly distributed between two provided durations, with millisecond precision
+     *
+     * @throws IllegalArgumentException if minimum duration is greater than maximum duration
+     */
+    public static Duration randomDuration(Duration minimum, Duration maximum)
+    {
+        Preconditions.checkArgument(minimum.compareTo(maximum) <= 0,
+                                    "Minimum duration must be less or equal to maximum duration");
+        return Duration.ofMillis(ThreadLocalRandom.current().nextLong(minimum.toMillis(), maximum.toMillis() + 1L));
```

Comment:
I would like to suggest an alternative. 

Instead of having `min` and `max` as the parameters of the method (and the implementation has to check that `min` is indeed not greater than `max`), you can pass `base` and `range`.
In addition to the implementation change, I do not see the necessity of using thread local random. A shared `Random` object should be suffice. You do not need to create a random object per calling thread. 

```suggestion
    public static Duration randomDuration(long base, long range)
    {
        long randomDurationMillis = base + random.nextLong() % range;
        return Duration.ofMillis(randomDurationMillis);
```

Reply:
I did make that change, but, to be honest, I didn't like what the code became.
Right now, "minimum duration" and "maximum duration" are easy to understand and explain.
With the change, "minimum duration" stays, but the actual meaning of the other value becomes confusing, at least to me.
"Maximum number of milliseconds to add on top of minimum duration" is a mouthful, and still needs to be checked for being non-negative.
Can we please keep the current implementation?

Reply:
ok.